plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -743,31 +764,44 @@ def finalize(self):
         self._pool.createwhatprovides()
         self._finalized = True
 
-    def load_repo(self, repo_id):
+    def load_source_repo(self, repo_id):
         """Load the provided Pulp repo as a source repo.
 
         All units in the repo will be available to be "installed", or copied.
         """
-        source_units = fetch_units_from_repo(repo_id)
-        self.add_repo_units(source_units, repo_id)
+        units = fetch_units_from_repo(repo_id)
+        self.add_repo_units(units, repo_id)
         _LOGGER.info('Loaded repository %s', repo_id)
 
-    def set_target_repo(self, repo_id):
-        """Load the provided Pulp repo as a target repo.
+    def load_target_repo(self, repo_id):
+        """Load the provided Pulp repo into the combined target repo.
 
-        All units in the repo are marked as "installed" inside libsolv, so that it knows they
-        don't need to be "installed" (copied)
+        All units in the repo will be added to the combined target repo, the contents of which
+        are considered "installed" by the solver.
         """
-        self._pool.installed = self.mapping.get_repo(str(repo_id))
-        _LOGGER.info('Target repository set to %s', repo_id)
+        units = fetch_units_from_repo(repo_id)
+        self.add_repo_units(units, repo_id, override_repo_name=COMBINED_TARGET_REPO_NAME)
```

Comment:
You seem to always use the `override_repo_name`. Do we need this additional parameter? Can we provide the name we need instead of `repo_id`? `repo_id` is used only to figure out the name, IIUC.

Reply:
I can go further :) Maybe we can just use this constant and not pass any parameter, if libsolv can have only one "installed" repo?

Reply:
done

Reply:
The reason is that ```add_repo_units``` is used in both ```load_source_repo``` and ```load_target_repo``` but only the latter function needs to redirect everything into a single repository.

Reply:
I see it know, last time I missed another call to `add_repo_units`.